lib/govuk_publishing_components/presenters/breadcrumb_selector.rb

```diff
@@ -0,0 +1,105 @@
+module GovukPublishingComponents
+  module Presenters
+    class BreadcrumbSelector
+      def self.call(*args)
+        new(*args).output
+      end
+
+      attr_reader :content_item, :request, :prioritise_taxon_breadcrumbs
+
+      def initialize(content_item, request, prioritise_taxon_breadcrumbs)
+        @content_item = content_item
+        @request = request
+        @prioritise_taxon_breadcrumbs = prioritise_taxon_breadcrumbs
+      end
+
+      def content_item_navigation
+        @content_item_navigation ||= ContextualNavigation.new(content_item, request)
+      end
+
+      def parent_item_navigation
+        @parent_item_navigation ||= ContextualNavigation.new(parent_item, request)
+      end
+
+      def parent_item
+        @parent_item ||= Services.content_store.content_item(content_item_navigation.parent_api_path)
+      rescue GdsApi::ContentStore::ItemNotFound
+        # Do nothing
+      end
+
+      def parent_item_options
+        @parent_item_options ||= options(parent_item_navigation)
+      end
+
+      def content_item_options
+        @content_item_options ||= options(content_item_navigation)
+      end
+
+      def parent_breadcrumbs
+        breadcrumbs = [parent_item_options[:breadcrumbs]].flatten # to ensure breadcrumbs always an array
+        breadcrumbs.last[:is_page_parent] = false
+        breadcrumbs << {
+          title: parent_item["title"],
+          url: parent_item["base_path"],
+          is_page_parent: true,
+        }
+      end
+
+      def output
+        return content_item_options unless content_item_navigation.html_document_with_parent?
+        return parent_item_options if parent_item_navigation.priority_breadcrumbs
+
+        {
+          step_by_step: parent_item_options[:step_by_step],
+          breadcrumbs: parent_breadcrumbs,
+        }
+      end
+
+      def options(navigation)
```

Comment:
Minor thing, but of the 8 branches in this method, 4 of them return the same thing: 
```
 {
    step_by_step: false,
    breadcrumbs: navigation.breadcrumbs,
  }
```

Maybe that is the default `option`?

Reply:
That mirrors the previous implementation. I didn't want to mess with the logic - just move it from a view partial into an object. Also, as the core of the logic is within the `options` method, I thought it clearer, and easier to modify if all the logical endpoints were of the same format.

Reply:
That makes sense.